File: 0101-symmetric-tree-对称二叉树/0101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def check(self, left, right):
        if not left or not right:
            return not left and not right
        if left.val != right.val:
            return False
        return self.check(left.left, right.right) and self.check(left.right, right.left)

    def isSymmetric(self, root: TreeNode) -> bool:
        if not root :
            return True
        return self.check(root.left, root.right)

File: 0101-symmetric-tree-对称二叉树/0101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
    bool DFS(TreeNode* left, TreeNode* right) {
        if (!left || !right) {
            return left == right;
        }
        return left->val == right->val && DFS(left->left, right->right) && DFS(left->right, right->left);
    }
public:
    bool isSymmetric(TreeNode* root) {
        if (!root) return true;
        return DFS(root->left, root->right);
    }
};